ci/infra/azure/master-instance.tf

```diff
@@ -3,6 +3,7 @@ resource "azurerm_network_interface" "master" {
   name                = "${var.stack_name}-master-${count.index}-nic"
   location            = azurerm_resource_group.resource_group.location
   resource_group_name = azurerm_resource_group.resource_group.name
+  depends_on          = [azurerm_subnet.nodes]
```

Comment:
Why do you need to mark this dependency in an explicit way? Should we update the existing v4 template too?

Reply:
Seems like there are race condition when `terraform apply/destroy` when use `cloud-init` instead of `virtual_machine_extesion`. If not declared explicitly, terraform script with stuck at some point in apply/destroy process.